src/db/add_package.rs

```diff
@@ -447,7 +447,7 @@ fn get_readme(pkg: &MetadataPackage, source_dir: &Path) -> Result<Option<String>
 }
 
 fn get_rustdoc(pkg: &MetadataPackage, source_dir: &Path) -> Result<Option<String>> {
-    if let Some(src_path) = &pkg.targets[0].src_path {
+    if let Some(src_path) = &pkg.targets.first().and_then(|t| t.src_path.clone()) {
```

Comment:
```suggestion
    if let Some(src_path) = &pkg.targets.first().map(|t| &t.src_path) {
```

Reply:
since `t.src_path` is an `Option<String>`, 

this would change the matched `src_path` from `&String` to `&Option<String>`

Reply:
What about:

```suggestion
    if let Some(src_path) = &pkg.targets.first().and_then(|t| &t.src_path) {
```

?

Reply:
nope :) 

<img width="873" alt="grafik" src="https://github.com/user-attachments/assets/950597f7-584c-4893-8e90-59d5f67a5cc0" />

But thinking about it, you're right, we didn't need the clone. Thanks for being persistent :) 

( added a commit)

Reply:
Haha. In my wrongness it allowed you to see a better path :D